spec/models/user_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.new(name: 'John', posts_counter: 2)
+    expect(user).to be_valid
+  end
+
+  it 'is not valid without a name' do
+    user = User.new(name: nil)
+    expect(user).to_not be_valid
+  end
+
+  it 'has a posts_counter greater than or equal to 0' do
+    user = User.new(posts_counter: -1)
+    expect(user).to_not be_valid
+  end
+
+  describe '#three_recent_posts' do
+    let(:user) { create(:user, name: 'John', posts_counter: 2) }
+
+    it 'returns the three most recent posts' do
+      old_post = create(:post, author: user, created_at: 3.days.ago)
+      recent_posts = create_list(:post, 3, author: user)
+
+      expect(user.three_recent_posts).to eq(recent_posts.reverse)
+      expect(user.three_recent_posts).not_to include(old_post)
+    end
+  end
```

Comment:
- [ ] You have done great writing the model specs and covering all the validations specs and custom methods, but looking at the screenshot below, you can see that only `3 tests` are passing the rest of the tests are failing. I'll suggest you fix these as all the tests are meant to pass. :+1: :smile: :wave: 

- - Screenshot:
![Screenshot from 2023-12-03 00-12-33](https://github.com/EvansSnave/Blog_app_rails/assets/107632780/1fac2a96-c981-4738-8a6d-0f70e254388c)